### ISSUE TITLE
Cannot provide post-condition specification to void methods 

### ISSUE THREAD

**Issue by leventeBajczi** [randoop/randoop#983]
Even though void methods do not return a value, it might be beneficial to test a postcondition after executing one.
For example, consider `Test.java`:
``` Java
public class Test{
    public static boolean tested = false;
    public void test() {
        tested = true;
    }
}
```
If I want to specify that `test()` always sets `Test.tested` to true, I could write the following `specification.json` file:
``` json
[ {
      "operation": {
        "classname": "Test",
        "name": "test",
        "parameterTypes": []
      },
      "identifiers": {
        "parameters": []
      },
      "post": [
        {
          "property": {
            "condition": "Test.tested"
          },
          "guard": {
            "condition": "true"
          },
          "description": "Test.test() sets Test.tested"
        }
      ]
} ]
```

However, test generation (via the command `java -cp $RANDOOP_JAR:. randoop.main.Main gentests --testclass Test --output-limit=2 --specifications=specification.json`, where `$RANDOOP_JAR` points to `randoop-all-4.2.6.jar`) throws the following exception:
``` java
Removing Test from the classes under test due to problem extracting operations:
randoop.condition.RandoopSpecificationError: Condition method did not compile:
2:55: 'void' type not allowed here

Class being compiled:
public class RandoopExpressionClass1 {
  public static boolean signature_Test(Test receiver, void result) throws Throwable {
    return Test.tested;
  }
}
```

This is expected, as `void` cannot appear as a parameter type declaration. For some time I thought the workaround would be to specify an empty string as the `returnName` in the `identifiers` block, but that will lead to an `OutOfMemoryError`, due to some string replacements (I reckon the empty string is replaced by some other string in all possible locations, which causes the error). The trace is the following:
``` java
java.lang.OutOfMemoryError: Java heap space
	at java.base/java.util.Arrays.copyOf(Arrays.java:3745)
	at java.base/java.lang.AbstractStringBuilder.ensureCapacityInternal(AbstractStringBuilder.java:172)
	at java.base/java.lang.AbstractStringBuilder.append(AbstractStringBuilder.java:538)
	at java.base/java.lang.StringBuilder.append(StringBuilder.java:174)
	at randoop.util.Util.replaceWords(Util.java:168)
	at randoop.condition.SpecificationTranslator.create(SpecificationTranslator.java:368)
	at randoop.condition.SpecificationTranslator.getReturnConditions(SpecificationTranslator.java:296)
	at randoop.condition.SpecificationTranslator.createExecutableSpecification(SpecificationTranslator.java:254)
	at randoop.condition.SpecificationCollection.getExecutableSpecification(SpecificationCollection.java:319)
	at randoop.reflection.OperationExtractor.visit(OperationExtractor.java:466)
	at randoop.reflection.ReflectionManager.applyTo(ReflectionManager.java:292)
	at randoop.reflection.ReflectionManager.apply(ReflectionManager.java:141)
	at randoop.reflection.OperationExtractor.operations(OperationExtractor.java:216)
	at randoop.reflection.OperationModel.addOperationsFromClasses(OperationModel.java:668)
	at randoop.reflection.OperationModel.createModel(OperationModel.java:172)
	at randoop.main.GenTests.handle(GenTests.java:317)
	at randoop.main.Main.nonStaticMain(Main.java:71)
	at randoop.main.Main.main(Main.java:31)
```

**To summarize**
1. void methods cannot appear in post-condition specifications
2. the empty string in an identifier causes an unhandled situation, which leads to an `OutOfMemoryError`
